tests/utils/utils.go

```diff
@@ -0,0 +1,190 @@
+	input erc20source.SendTokensInput,
+	amount *big.Int,
+	senderKey *ecdsa.PrivateKey,
+) (*types.Receipt, [32]byte, *big.Int) {
+	// Approve the ERC20Source to spend the tokens
+	teleporterUtils.ERC20Approve(
+		ctx,
+		sourceToken,
+		erc20SourceAddress,
+		amount,
+		subnet,
+		senderKey,
+	)
+
+	// Send the tokens and verify expected events
+	optsA, err := bind.NewKeyedTransactorWithChainID(senderKey, subnet.EVMChainID)
+	Expect(err).Should(BeNil())
+	tx, err := erc20Source.Send(
+		optsA,
+		input,
+		amount,
+	)
+	Expect(err).Should(BeNil())
+	bridgedAmount := big.NewInt(0).Sub(amount, input.PrimaryFee)
+
+	receipt := teleporterUtils.WaitForTransactionSuccess(ctx, subnet, tx.Hash())
+	event, err := teleporterUtils.GetEventFromLogs(receipt.Logs, erc20Source.ParseSendTokens)
+	Expect(err).Should(BeNil())
+	Expect(event.Sender).Should(Equal(crypto.PubkeyToAddress(senderKey.PublicKey)))
+	Expect(event.Amount).Should(Equal(bridgedAmount))
+
+	return receipt, event.TeleporterMessageID, event.Amount
+}
+
+func SendERC20Destination(
+	ctx context.Context,
+	subnet interfaces.SubnetTestInfo,
+	erc20Destination *erc20destination.ERC20Destination,
+	erc20DestinationAddress common.Address,
+	input erc20destination.SendTokensInput,
+	amount *big.Int,
+	senderKey *ecdsa.PrivateKey,
+) (*types.Receipt, [32]byte, *big.Int) {
+	opts, err := bind.NewKeyedTransactorWithChainID(senderKey, subnet.EVMChainID)
+	Expect(err).Should(BeNil())
+	tx, err := erc20Destination.Approve(
+		opts,
+		erc20DestinationAddress,
+		amount,
+	)
+	Expect(err).Should(BeNil())
+
+	teleporterUtils.WaitForTransactionSuccess(ctx, subnet, tx.Hash())
+
+	// Bridge the tokens back to subnet A
+	tx, err = erc20Destination.Send(
+		opts,
+		input,
+		amount,
+	)
+	Expect(err).Should(BeNil())
+
+	bridgedAmount := big.NewInt(0).Sub(amount, input.PrimaryFee)
+	receipt := teleporterUtils.WaitForTransactionSuccess(ctx, subnet, tx.Hash())
+	event, err := teleporterUtils.GetEventFromLogs(receipt.Logs, erc20Destination.ParseSendTokens)
+	Expect(err).Should(BeNil())
+	Expect(event.Sender).Should(Equal(crypto.PubkeyToAddress(senderKey.PublicKey)))
+	Expect(event.Amount).Should(Equal(bridgedAmount))
+
+	return receipt, event.TeleporterMessageID, event.Amount
+}
+
+func CheckERC20SourceWithdrawal(
+	ctx context.Context,
+	erc20SourceAddress common.Address,
+	sourceToken *exampleerc20.ExampleERC20,
+	recipientAddress common.Address,
+	amount *big.Int,
+	receipt *types.Receipt,
+) {
+	sourceTransferEvent, err := teleporterUtils.GetEventFromLogs(receipt.Logs, sourceToken.ParseTransfer)
+	Expect(err).Should(BeNil())
+	Expect(sourceTransferEvent.From).Should(Equal(erc20SourceAddress))
+	Expect(sourceTransferEvent.To).Should(Equal(recipientAddress))
+	Expect(sourceTransferEvent.Value).Should(Equal(amount))
+}
+
+func CheckERC20DestinationWithdrawal(
+	ctx context.Context,
+	erc20Destination *erc20destination.ERC20Destination,
+	recipientAddress common.Address,
+	amount *big.Int,
```

Comment:
May want to consider naming these "expectedRecipientAddress" and "expectedAmount". (Same for check source withdrawal above)